# **Inteligencia Artificial con Aplicaciones en Economía I**

- 👩‍🏫 **Profesora:** [Lina María Castro](https://www.linkedin.com/in/lina-maria-castro)  
- 📧 **Email:** [lmcastroco@gmail.com](mailto:lmcastroco@gmail.com)  
- 🎓 **Universidad:** Universidad Externado de Colombia - Facultad de Economía

# **Taller Limpieza y transformación de datos**🧹

**IMPORTANTE**: Guarda una copia de este notebook en tu Google Drive o computador.

**Taller en parejas.**

**Nombres estudiantes:**

-
-

**Forma de entrega:**

- Nombrar el archivo de la siguiente forma:“Taller_Limpieza_apellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.cloud.microsoft/r/8Qw28N1t1j.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

**Plazo de entrega:**

8 de septiembre de 2026, máximo a las 11:59 p.m. Tenga en cuenta que luego de esa hora el formulario en forms se cierra. El Jupupyter Notebook también debe quedar subido en Github antes de esa hora.

**Instrucciones Generales:**

Completa el código en las celdas marcadas con `### TU CÓDIGO AQUÍ ###`. Puedes añadir más celdas si lo requieres.

Utilice la base de la GEIH de enero a junio de 2026, la cual se encuentra en el repositorio en Github del curso o en Google Drive en la carpeta "datasets". Tenga en cuenta sep=";" y encoding='latin-1'. Las variables 'DPTO' y 'RAMA4D_R4' deben ser cargadas como string.

# Ejercicio 1: Importar librerías y cargar datos

1.	Importe todas las librerías que vaya a utilizar.

In [1]:
### TU CÓDIGO AQUÍ ###
import pandas as pd
import numpy as np
import os
import warnings
warnings.filterwarnings('ignore')

2.	Mejore la visualización de los dataframes como se ha visto en clase.

In [2]:
### TU CÓDIGO AQUÍ ###
# Que muestre todas las columnas
pd.options.display.max_columns = None
# En los dataframes, mostrar los float con dos decimales
pd.options.display.float_format = '{:,.2f}'.format

3.	Conéctese a su Google Drive y establezca la ruta del dataset.

In [5]:
### TU CÓDIGO AQUÍ ###
from google.colab import drive, files
drive.mount('/content/drive')
ruta = '/content/drive/MyDrive/Colab Notebooks/GEIH'
os.chdir(ruta)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


4.	Cargue los datos en el notebook. Llame al dataframe 'df_geih'

In [6]:
### TU CÓDIGO AQUÍ ###
df_geih = pd.read_csv('geih_2026_completa.csv')

5. Obtenga las dimensiones de la base

In [7]:
### TU CÓDIGO AQUÍ ###
df_geih.shape

(401950, 321)

In [13]:
df_geih.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 401950 entries, 0 to 401949
Columns: 321 entries, PERIODO to NOMBRE_DEPARTAMENTO
dtypes: float64(251), int64(46), object(24)
memory usage: 984.4+ MB


# Ejercicio 2:  Seleccionar y renombrar columnas

La GEIH contiene cientos de variables. Para nuestro análisis del mercado laboral, nos enfocaremos en un subconjunto.

Adicionalmente, los nombres de las columnas en la GEIH son códigos. Para hacer el análisis más intuitivo, vamos a renombrarlas.

1. Crea un nuevo DataFrame llamado `df_empleo` que contenga únicamente las columnas que están en la lista `columnas_interes`.

In [14]:
columnas_interes = ['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'MES', 'CLASE', 'DPTO', 'FEX_C18', 'PT', 'P3271', 'P6040', 'P6080', 'P3043', 'OCI',
                    'P6430', 'RAMA4D_R4', 'INGLABO', 'DESCRIPCION_CIIU', 'NOMBRE_DEPARTAMENTO']

In [58]:
### TU CÓDIGO AQUÍ ###
df_empleo = df_geih[columnas_interes]
df_empleo.head()

,DIRECTORIO,SECUENCIA_P,ORDEN,MES,CLASE,DPTO,FEX_C18,PT,P3271,P6040,P6080,P3043,OCI,P6430,RAMA4D_R4,INGLABO,DESCRIPCION_CIIU,NOMBRE_DEPARTAMENTO
0,8473909,1,1,1,2,8,41.34,1,2,34,6,NaN,1.00,4.00,"4,792.00","100,000.00",Comercio al por menor realizado a través de ca...,ATLÁNTICO
1,8473909,1,2,1,2,8,41.34,1,1,49,6,NaN,1.00,4.00,"9,522.00","1,900,000.00",Mantenimiento y reparación de aparatos y equip...,ATLÁNTICO
2,8473909,1,3,1,2,8,41.34,1,2,17,6,1.00,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
3,8473909,1,4,1,2,8,41.34,1,1,15,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
4,8473909,1,5,1,2,8,41.34,1,1,12,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO


2. Renombra las columnas del DataFrame `df_empleo` utilizando el diccionario `mapa_nombres`.

In [59]:
mapa_nombres = {
'DIRECTORIO': 'id_vivienda',
'SECUENCIA_P': 'id_hogar',
'ORDEN': 'id_persona',
'MES': 'mes',
'CLASE': 'clase',
'DPTO': 'departamento',
'FEX_C18': 'factor_expansion',
'PT': 'poblacion_total',
'P3271': 'sexo',
'P6040': 'edad',
'P6080': 'grupo_etnico',
'P3043': 'nivel_educativo',
'OCI': 'ocupados',
'P6430': 'categoria_ocupacional',
'RAMA4D_R4': 'cod_ciiu',
'INGLABO': 'ingreso_laboral',
'DESCRIPCION_CIIU': 'desc_ciiu',
'NOMBRE_DEPARTAMENTO': 'nombre_dpto'}

In [60]:
### TU CÓDIGO AQUÍ ###
df_empleo.rename(columns=mapa_nombres, inplace=True)
df_empleo.head()

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
0,8473909,1,1,1,2,8,41.34,1,2,34,6,NaN,1.00,4.00,"4,792.00","100,000.00",Comercio al por menor realizado a través de ca...,ATLÁNTICO
1,8473909,1,2,1,2,8,41.34,1,1,49,6,NaN,1.00,4.00,"9,522.00","1,900,000.00",Mantenimiento y reparación de aparatos y equip...,ATLÁNTICO
2,8473909,1,3,1,2,8,41.34,1,2,17,6,1.00,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
3,8473909,1,4,1,2,8,41.34,1,1,15,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
4,8473909,1,5,1,2,8,41.34,1,1,12,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO


3. Obtenga las dimensiones de la base

In [61]:
### TU CÓDIGO AQUÍ ###
df_empleo.shape

(401950, 18)

# Ejercicio 3: Revisión de duplicados

Un paso crucial en la limpieza es asegurar que no haya observaciones duplicadas que puedan sesgar nuestro análisis.

1. Verifica si hay filas duplicadas en `df_empleo`.

In [62]:
### TU CÓDIGO AQUÍ ###
df_empleo[df_empleo.duplicated()]

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto


2. Cuenta cuántas filas duplicadas existen.

In [63]:
### TU CÓDIGO AQUÍ ###
df_empleo.duplicated().sum()

np.int64(0)

3. Elimina las filas duplicadas, si las hay.

In [ ]:
### TU CÓDIGO AQUÍ ###


4. Verifica si hay filas duplicadas específicamente en las columnas 'id_vivienda'. Si hay duplicados, explica por qué existen estos duplicados en la GEIH para esta columna.

In [64]:
### TU CÓDIGO AQUÍ ###
df_empleo[df_empleo.duplicated(subset=['id_vivienda'])]

### En este caso existen datos duplicados en la columna 'id_vivienda'
# Ya que en un hogar viven varias personas, y la columna identifica que
#personas viven en el mismo hogar

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,ingreso_laboral,desc_ciiu,nombre_dpto
1,8473909,1,2,1,2,8,41.34,1,1,49,6,NaN,1.00,4.00,"9,522.00","1,900,000.00",Mantenimiento y reparación de aparatos y equip...,ATLÁNTICO
2,8473909,1,3,1,2,8,41.34,1,2,17,6,1.00,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
3,8473909,1,4,1,2,8,41.34,1,1,15,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
4,8473909,1,5,1,2,8,41.34,1,1,12,6,NaN,NaN,NaN,NaN,NaN,NaN,ATLÁNTICO
6,8473910,1,2,1,2,8,43.56,1,2,26,6,NaN,1.00,4.00,"9,602.00","200,000.00",Peluquería y otros tratamientos de belleza,ATLÁNTICO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
401942,8653999,3,2,6,1,97,7.28,1,2,17,1,1.00,NaN,NaN,NaN,NaN,NaN,VAUPÉS
401943,8653999,3,3,6,1,97,7.28,1,2,15,1,NaN,NaN,NaN,NaN,NaN,NaN,VAUPÉS
401945,8654001,1,2,6,1,97,15.47,1,2,9,1,NaN,NaN,NaN,NaN,NaN,NaN,VAUPÉS
401946,8654001,1,3,6,1,97,15.47,1,1,3,1,NaN,NaN,NaN,NaN,NaN,NaN,VAUPÉS


# Ejercicio 4: Valores faltantes

Los datos del mundo real raramente están completos. Aquí, trataremos los valores faltantes en la columna de ingresos.


1. Calcula el número de valores nulos (`NaN`) en la columna `ingreso_laboral` **antes** de cualquier modificación.

In [65]:
### TU CÓDIGO AQUÍ ###
df_empleo['ingreso_laboral'].isna().sum()

np.int64(228153)

2. En la GEIH, un ingreso faltante a menudo significa que la persona no reportó ingresos o no aplica (ej. no trabaja). Una estrategia común es imputar estos valores con `0`. Rellena los valores `NaN` en la columna `ingreso_laboral` con `0`.

In [66]:
### TU CÓDIGO AQUÍ ###
df_empleo['ingreso_laboral'].fillna(0)

,ingreso_laboral
0,"100,000.00"
1,"1,900,000.00"
2,0.00
3,0.00
4,0.00
...,...
401945,0.00
401946,0.00
401947,"3,000,000.00"
401948,0.00


# Ejercicio 5: Decodificar variables



Las variables categóricas a menudo necesitan ser "traducidas" para hacerlas más entendibles para el análisis.

1. Usa los siguientes diccionarios para reemplazar los códigos numéricos por etiquetas de texto en las siguientes columnas: 'clase', 'sexo', 'grupo_etnico', 'nivel_educativo' y 'categoria_ocupacional'. Muestra los conteos de las nuevas categorías usando `.value_counts()` para verificar el cambio en cada una de las columnas.

In [67]:
dic_clase = {1: 'Cabeceras', 2: 'Centros poblados y rural disperso'}
dic_sexo = {1: 'Hombre', 2: 'Mujer'}
dic_etnico = {1: 'Indígena', 2: 'Gitano (Rom)', 3: 'Raizal del archipiélago', 4: 'Palenquero',
              5: 'Negro, mulato (afrodescendiente)', 6: 'Ninguno de los anteriores'}
dic_nivel_educativo = {1: 'Ninguno', 2: 'Educación media', 3: 'Educación media',
                       4: 'Educación media', 5: 'Técnica profesional', 6: 'Tecnológica',
                       7: 'Universitaria', 8: 'Especialización', 9: 'Maestría',
                       10: 'Doctorado', 99: 'No sabe, no informa'}
dic_cat_ocu = {1: 'Asalariado', 2: 'Asalariado', 3: 'Asalariado',
               8: 'Asalariado', 4: 'Independiente', 5: 'Independiente',
               9: 'Independiente', 6: 'Trabajador sin remuneración',
               7: 'Trabajador sin remuneración'}

In [68]:
### TU CÓDIGO AQUÍ ###
df_empleo['clase'] = df_empleo['clase'].map(dic_clase)
print(df_empleo['clase'].value_counts())

clase
Cabeceras                            348316
Centros poblados y rural disperso     53634
Name: count, dtype: int64


In [69]:
### TU CÓDIGO AQUÍ ###
df_empleo['sexo'] = df_empleo['sexo'].map(dic_sexo)
print(df_empleo['sexo'].value_counts())

sexo
Mujer     214629
Hombre    187321
Name: count, dtype: int64


In [70]:
### TU CÓDIGO AQUÍ ###
df_empleo['grupo_etnico'] = df_empleo['grupo_etnico'].map(dic_etnico)
print(df_empleo['grupo_etnico'].value_counts())

grupo_etnico
Ninguno de los anteriores           345317
Negro, mulato (afrodescendiente)     35191
Indígena                             20331
Raizal del archipiélago                971
Palenquero                              90
Gitano (Rom)                            50
Name: count, dtype: int64


In [71]:
### TU CÓDIGO AQUÍ ###
df_empleo['nivel_educativo'] = df_empleo['nivel_educativo'].map(dic_nivel_educativo)
print(df_empleo['nivel_educativo'].value_counts())

nivel_educativo
Educación media        114830
Universitaria           35105
Técnica profesional     24188
Ninguno                 12408
Tecnológica             10651
Especialización          9618
Maestría                 4091
Doctorado                 423
Name: count, dtype: int64


In [72]:
### TU CÓDIGO AQUÍ ###
df_empleo['categoria_ocupacional'] = df_empleo['categoria_ocupacional'].map(dic_cat_ocu)
print(df_empleo['categoria_ocupacional'].value_counts())

categoria_ocupacional
Asalariado                     90891
Independiente                  79620
Trabajador sin remuneración     6684
Name: count, dtype: int64


# Ejercicio 6: Reorganizar columnas

1. Reorganiza las columnas con el siguiente orden: 'id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto',     'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral'

In [75]:
### TU CÓDIGO AQUÍ ###
nuevo_orden = ['id_vivienda', 'id_hogar', 'id_persona', 'mes', 'clase', 'departamento', 'nombre_dpto', 'factor_expansion', 'poblacion_total', 'sexo', 'edad', 'grupo_etnico', 'nivel_educativo', 'ocupados', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu', 'ingreso_laboral']
df_empleo = df_empleo[nuevo_orden]
df_empleo.head()

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
0,8473909,1,1,1,Centros poblados y rural disperso,8,ATLÁNTICO,41.34,1,Mujer,34,Ninguno de los anteriores,NaN,1.00,Independiente,"4,792.00",Comercio al por menor realizado a través de ca...,"100,000.00"
1,8473909,1,2,1,Centros poblados y rural disperso,8,ATLÁNTICO,41.34,1,Hombre,49,Ninguno de los anteriores,NaN,1.00,Independiente,"9,522.00",Mantenimiento y reparación de aparatos y equip...,"1,900,000.00"
2,8473909,1,3,1,Centros poblados y rural disperso,8,ATLÁNTICO,41.34,1,Mujer,17,Ninguno de los anteriores,Ninguno,NaN,NaN,NaN,NaN,NaN
3,8473909,1,4,1,Centros poblados y rural disperso,8,ATLÁNTICO,41.34,1,Hombre,15,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
4,8473909,1,5,1,Centros poblados y rural disperso,8,ATLÁNTICO,41.34,1,Hombre,12,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN


# Ejercicio 7: Cambiar tipo de dato


Para realizar cálculos y análisis correctamente, las columnas deben tener el tipo de dato adecuado (numérico, categórico, etc.).

1.  Muestra en pantalla los tipos de datos actuales de `df_empleo`.

In [76]:
### TU CÓDIGO AQUÍ ###
df_empleo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 401950 entries, 0 to 401949
Data columns (total 18 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   id_vivienda            401950 non-null  int64  
 1   id_hogar               401950 non-null  int64  
 2   id_persona             401950 non-null  int64  
 3   mes                    401950 non-null  int64  
 4   clase                  401950 non-null  object 
 5   departamento           401950 non-null  int64  
 6   nombre_dpto            401950 non-null  object 
 7   factor_expansion       401950 non-null  float64
 8   poblacion_total        401950 non-null  int64  
 9   sexo                   401950 non-null  object 
 10  edad                   401950 non-null  int64  
 11  grupo_etnico           401950 non-null  object 
 12  nivel_educativo        211314 non-null  object 
 13  ocupados               177195 non-null  float64
 14  categoria_ocupacional  177195 non-nu

2.  Las siguientes columnas representan categorías: 'clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu'. Conviértelas al tipo `category` para optimizar el uso de memoria y facilitar análisis futuros.

In [80]:
### TU CÓDIGO AQUÍ ###
categorias= [ 'clase', 'departamento', 'nombre_dpto', 'sexo', 'grupo_etnico', 'nivel_educativo', 'categoria_ocupacional', 'cod_ciiu', 'desc_ciiu']
for categoria in categorias:
  df_empleo[categoria]  = df_empleo[categoria].astype('category')
df_empleo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 401950 entries, 0 to 401949
Data columns (total 18 columns):
 #   Column                 Non-Null Count   Dtype   
---  ------                 --------------   -----   
 0   id_vivienda            401950 non-null  int64   
 1   id_hogar               401950 non-null  int64   
 2   id_persona             401950 non-null  int64   
 3   mes                    401950 non-null  int64   
 4   clase                  401950 non-null  category
 5   departamento           401950 non-null  category
 6   nombre_dpto            401950 non-null  category
 7   factor_expansion       401950 non-null  float64 
 8   poblacion_total        401950 non-null  int64   
 9   sexo                   401950 non-null  category
 10  edad                   401950 non-null  int64   
 11  grupo_etnico           401950 non-null  category
 12  nivel_educativo        211314 non-null  category
 13  ocupados               177195 non-null  float64 
 14  categoria_ocupaciona

# Ejercicio 8: Ordenar el dataframe (5%)

1. Ordena el dataframe por la columna "edad" de mayor a menor. Observa la columna 'edad' y escribe en una celda de markdown cuál fue la edad mínima y la edad máxima de las personas que respondieron la encuesta de la GEIH.

In [81]:
### TU CÓDIGO AQUÍ ###
df_empleo.sort_values(by='edad', ascending=False)

,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
354475,8631567,1,6,6,Centros poblados y rural disperso,13,BOLÍVAR,97.31,1,Mujer,107,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
51187,8497328,1,3,1,Cabeceras,20,CESAR,166.26,1,Mujer,107,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
356371,8632352,1,2,6,Cabeceras,68,SANTANDER,743.70,1,Mujer,107,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
276488,8599132,1,11,5,Cabeceras,20,CESAR,534.29,1,Hombre,105,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
20775,8482658,1,1,1,Cabeceras,52,NARIÑO,155.88,1,Mujer,105,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
334990,8623565,1,6,5,Cabeceras,85,CASANARE,389.64,1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
46461,8495404,1,3,1,Cabeceras,68,SANTANDER,475.48,1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
14677,8480084,1,6,1,Centros poblados y rural disperso,41,HUILA,"2,404.05",1,Hombre,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN
240643,8584082,1,4,4,Cabeceras,8,ATLÁNTICO,788.05,1,Mujer,0,Ninguno de los anteriores,NaN,NaN,NaN,NaN,NaN,NaN


Escribe tu análisis aquí:


In [84]:
from IPython.core.display import Markdown

La edad maxima de las personas que respondieron la encuesta es de 107 y la edad minima es de 0

# Ejercicio 9: Groupby

Agrupar los datos por categorías, nos permite realizar análisis más profundos y obtener insights valiosos, que a simple vista no vemos en la base de datos.

1. Crea un nuevo dataframe llamado "df_sexo" donde se agrupe por 'sexo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de sexo en una columna.

In [87]:
### TU CÓDIGO AQUÍ ###
df_sexo = df_empleo.groupby('sexo').agg(suma_poblacion_total=('poblacion_total', 'sum'), suma_ocupados=('ocupados', 'sum')).reset_index()
df_sexo['porcentaje_ocupados'] = df_sexo['suma_ocupados'] / df_sexo['suma_poblacion_total']
df_sexo.head()

,sexo,suma_poblacion_total,suma_ocupados,porcentaje_ocupados
0,Hombre,187321,"98,214.00",0.52
1,Mujer,214629,"78,981.00",0.37


2. Crea un nuevo dataframe llamado "df_educacion" donde se agrupe por 'nivel_educativo' y se calcule la suma de las columnas 'poblacion_total' y 'ocupados'. Crea una nueva columna que se llame 'porcentaje_ocupados' donde dividas las columna 'ocupados' sobre 'poblacion_total'. Al final usa reset_index() para convertir las categorías de nivel educativo en una columna.

In [88]:
### TU CÓDIGO AQUÍ ###
df_educacion =df_empleo.groupby('nivel_educativo').agg(
    suma_poblacion_total=('poblacion_total','sum'), suma_ocupados=('ocupados','sum')).reset_index()
df_educacion['porcentaje_ocupados'] = df_educacion['suma_ocupados'] / df_educacion['suma_poblacion_total']
df_educacion.head()


,nivel_educativo,suma_poblacion_total,suma_ocupados,porcentaje_ocupados
0,Doctorado,423,357.00,0.84
1,Educación media,114830,"64,287.00",0.56
2,Especialización,9618,"7,524.00",0.78
3,Maestría,4091,"3,458.00",0.85
4,Ninguno,12408,"2,271.00",0.18


3. En una celda de Markdown escribe qué conclusiones puedes sacar de la información obtenida en df_sexo y df_educacion.

Escribe tu análisis aquí: Las mujeres representan el mayor numero de personas encuestadas, sin embargo son las que se encuentran menos ocupadas. Ademas la mayor parte de la poblacion tiene una educacion media y asi mismo es el nivel educativo que tiene un mayor numero de ocupados.

# Ejercicio 10: Filtrar datos

1. A partir de df_empleo, crea el DataFrame `df_ing` con observaciones donde `ingreso_laboral` no sea nulo y además sea mayor estrictamente al salario mínimo para 2026 (con subsidio de transporte).

In [107]:
### TU CÓDIGO AQUÍ ###
salario_minimo_2026 = 2000000

df_ing = df_empleo[
    (df_empleo['ingreso_laboral'].notna()) &
    (df_empleo['ingreso_laboral'] > salario_minimo_2026)
].reset_index(drop=True)

print(df_ing.shape)
df_ing

(50103, 18)


,id_vivienda,id_hogar,id_persona,mes,clase,departamento,nombre_dpto,factor_expansion,poblacion_total,sexo,edad,grupo_etnico,nivel_educativo,ocupados,categoria_ocupacional,cod_ciiu,desc_ciiu,ingreso_laboral
0,8473932,1,2,1,Centros poblados y rural disperso,13,BOLÍVAR,"1,304.56",1,Hombre,56,"Negro, mulato (afrodescendiente)",Universitaria,1.00,Independiente,"6,910.00",Actividades jurídicas,"6,000,000.00"
1,8473945,1,2,1,Cabeceras,44,LA GUAJIRA,181.19,1,Hombre,58,Ninguno de los anteriores,Educación media,1.00,Asalariado,"4,111.00",Construcción de edificios residenciales,"2,720,000.00"
2,8473948,1,1,1,Cabeceras,44,LA GUAJIRA,294.36,1,Hombre,41,Ninguno de los anteriores,Educación media,1.00,Independiente,"4,781.00","Comercio al por menor de alimentos, bebidas y ...","2,100,000.00"
3,8473954,1,2,1,Cabeceras,44,LA GUAJIRA,280.31,1,Mujer,56,Ninguno de los anteriores,Especialización,1.00,Asalariado,"8,513.00",Educación básica primaria,"2,800,000.00"
4,8473959,1,1,1,Cabeceras,44,LA GUAJIRA,520.46,1,Hombre,47,Indígena,Maestría,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"3,500,000.00"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
50098,8653996,1,1,6,Cabeceras,94,GUAINÍA,50.52,1,Hombre,60,Ninguno de los anteriores,Tecnológica,1.00,Asalariado,"8,413.00",Regulación de las actividades de organismos qu...,"2,600,000.00"
50099,8653997,1,2,6,Cabeceras,95,GUAVIARE,125.10,1,Hombre,38,Ninguno de los anteriores,Tecnológica,1.00,Independiente,"3,513.00",Distribución de energía eléctrica,"3,100,000.00"
50100,8653999,1,1,6,Cabeceras,97,VAUPÉS,7.28,1,Hombre,42,Indígena,Técnica profesional,1.00,Independiente,"8,412.00",Actividades ejecutivas de la administración pú...,"3,200,000.00"
50101,8654001,1,1,6,Cabeceras,97,VAUPÉS,15.47,1,Mujer,38,Indígena,Técnica profesional,1.00,Asalariado,"8,530.00",Establecimientos que combinan diferentes nivel...,"2,100,000.00"


2. ¿Cuántas personas tienen un salario estrictamente mayor al salario mínimo?

In [100]:
### TU CÓDIGO AQUÍ ###
df_ing[df_ing['ingreso_laboral']>salario_minimo_2026]
print(df_ing.shape)

(50103, 18)


3. Crea el dataframe df_mujeres donde se encuentren solamente mujeres que están ubicadas en el departamento del Tolima, que son asalariadas y cuyo nivel educativo es educación media o superior.

In [115]:
### TU CÓDIGO AQUÍ ###
df_mujeres = df_empleo[
    (df_empleo['sexo']=='Mujer') &
    (df_empleo['nombre_dpto']=='TOLIMA') &
    (df_empleo['categoria_ocupacional']=='Asalariado') &
    (df_empleo['nivel_educativo'].isin(['Educación media', 'Técnica profesional', 'Universitaria', 'Especialización', 'Maestría', 'Tecnológica', 'Doctorado']))
]
df_mujeres.shape

(1044, 18)

4. ¿Cuáles son las dimensiones de df_mujeres?

In [119]:
### TU CÓDIGO AQUÍ ###
df_mujeres.shape
df_mujeres.info()


<class 'pandas.core.frame.DataFrame'>
Index: 1044 entries, 9390 to 401880
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype   
---  ------                 --------------  -----   
 0   id_vivienda            1044 non-null   int64   
 1   id_hogar               1044 non-null   int64   
 2   id_persona             1044 non-null   int64   
 3   mes                    1044 non-null   int64   
 4   clase                  1044 non-null   category
 5   departamento           1044 non-null   category
 6   nombre_dpto            1044 non-null   category
 7   factor_expansion       1044 non-null   float64 
 8   poblacion_total        1044 non-null   int64   
 9   sexo                   1044 non-null   category
 10  edad                   1044 non-null   int64   
 11  grupo_etnico           1044 non-null   category
 12  nivel_educativo        1044 non-null   category
 13  ocupados               1044 non-null   float64 
 14  categoria_ocupacional  1044 non-null   c